src/lib/rule2F.ts

```diff
@@ -138,6 +138,93 @@ function allowsNameFromContent(elem: HTMLElement): boolean {
 
 export const TEST_ONLY = {allowsNameFromContent};
 
+// See https://lists.w3.org/Archives/Public/public-aria/2017Jun/0057.html
+// for discussion of roles & tags that forbid name from content.
+const NO_NAME_FROM_CONTENT_ROLES = [
```

Comment:
I think `NAME_FROM_CONTENT_FORBIDDEN_ROLES` or `NEVER_NAME_FROM_CONTENT_ROLES` makes it more clear (especially when you add conditional name from content roles)